feat: new features

VeeverSW opened the pull request:
1. 增加限制，避免了重复申请正在排的副本的无法连接到任务搜索服务器问题
2. 新增vnavmesh寻路方法，在主窗口中新增寻路方式切换选项
3. 运行设置中增加了停止寻路按钮（避免突发状况）
4. 在调试窗口中增加了鼠标位置转换，此窗口打开时，可使用ctrl+c复制当前鼠标指针所在坐标
5. 在预设-坐标中增加了粘贴坐标，支持第四条的鼠标位置转换复制得到的坐标格式
6. 在预设-坐标中增加了寻路到此位置按钮

VeeverSW committed:
feat: new features
1. 增加限制，避免了重复申请正在排的副本的无法连接到任务搜索服务器问题
2. 删除了原生movePath方法，新增vnavIPC的调用来处理寻路的问题
3. 运行设置中增加了停止寻路按钮（避免突发状况）
4. 在调试窗口中增加了鼠标位置转换，此窗口打开时，可使用ctrl+c复制当前鼠标指针所在坐标
5. 在预设-坐标中增加了粘贴坐标，支持第四条的鼠标位置转换复制得到的坐标格式
6. 在预设-坐标中增加了寻路到此位置按钮

AtmoOmen reviewed (changes requested):
不能删除原寻路方法，只能做加法，需要向前兼容

VeeverSW committed:
feat: new features
1. 增加限制，避免了重复申请正在排的副本的无法连接到任务搜索服务器问题
2. 新增vnavmesh寻路方法，在主窗口中新增寻路方式切换选项
3. 运行设置中增加了停止寻路按钮（避免突发状况）
4. 在调试窗口中增加了鼠标位置转换，此窗口打开时，可使用ctrl+c复制当前鼠标指针所在坐标
5. 在预设-坐标中增加了粘贴坐标，支持第四条的鼠标位置转换复制得到的坐标格式
6. 在预设-坐标中增加了寻路到此位置按钮

VeeverSW committed:
feat: new features
1. 增加限制，避免了重复申请正在排的副本的无法连接到任务搜索服务器问题
2. 新增vnavmesh寻路方法，在主窗口中新增寻路方式切换选项
3. 运行设置中增加了停止寻路按钮（避免突发状况）
4. 在调试窗口中增加了鼠标位置转换，此窗口打开时，可使用ctrl+c复制当前鼠标指针所在坐标
5. 在预设-坐标中增加了粘贴坐标，支持第四条的鼠标位置转换复制得到的坐标格式
6. 在预设-坐标中增加了寻路到此位置按钮

VeeverSW committed:
feat: delete useless using

AtmoOmen commented on UntarnishedHeart/Utils/GameFunctions.cs:
```diff
@@ -62,17 +93,91 @@ public static unsafe void Teleport(Vector3 pos)
+        while (true)
+        {
+            var localPlayer = DService.ObjectTable.LocalPlayer;
+            if (localPlayer == null) continue;
```
is 匹配

AtmoOmen reviewed (changes requested):
1. vnavmesh 写成独立于传送、寻路以外的第三个Mode，而不是单独再开一个寻路模式
2. 命名为 `vnavmeshIPC`
3. 日志打印全部使用 OmenTools 中 NotifyHelper 内置的方法

VeeverSW committed:
fix:
1. 修改所有的命名(VNavmesh->vnavmesh)
2. 修复了vnavmeshStopIPC的调用
3. 添加了继续寻路按钮
4. 独立vnavmesh寻路
5. 删除寻路模式切换
6. 将传送到此位置按钮整合成根据当前移动方式移动到此位置

AtmoOmen commented on UntarnishedHeart/Windows/Main.cs:
```diff
@@ -275,6 +276,22 @@ private static void DrawHomeExecutorConfig()
+        }
+        ImGuiOm.TooltipHover("立即停止当前的寻路任务");
+
+
```
目前仅在简单路线模式下添加，但实则应该通用，目前设计的 UI 位置分类也存在混淆的可能性

AtmoOmen commented on UntarnishedHeart/Windows/Debug.cs:
```diff
@@ -181,18 +195,76 @@ private static void DrawDebugStatusInfo()
+            ImGui.Spacing();
+            
+            // 17 = CONTROL, 67 = 'C'
+            if (DService.KeyState[0x11] && DService.KeyState[0x43] &&
```
使用枚举

AtmoOmen commented on UntarnishedHeart/Windows/Debug.cs:
```diff
@@ -181,18 +195,76 @@ private static void DrawDebugStatusInfo()
+            
+            // 17 = CONTROL, 67 = 'C'
+            if (DService.KeyState[0x11] && DService.KeyState[0x43] &&
+                (DateTime.Now - lastCopyTime).TotalSeconds > 0.5)
```
使用系统刻, DateTime 对象开销大

AtmoOmen commented on UntarnishedHeart/Windows/Debug.cs:
```diff
@@ -181,18 +195,76 @@ private static void DrawDebugStatusInfo()
+                NotifyHelper.NotificationSuccess($"已复制坐标到剪贴板: <{worldPos.X:F2}, {worldPos.Y:F2}, {worldPos.Z:F2}>");
+            }
+
+            if (ImGui.Button("Ctrl+C 复制坐标"))
```
`复制 (Ctrl + C)`

AtmoOmen commented on UntarnishedHeart/Windows/Debug.cs:
```diff
@@ -181,18 +195,76 @@ private static void DrawDebugStatusInfo()
+        {
+            ImGui.TextColored(ImGuiColors.DalamudYellow, "鼠标未指向游戏世界");
+            ImGui.TextWrapped("提示: 将鼠标移动到游戏世界中的物体或地面上");
+        }
```
删除

AtmoOmen commented on UntarnishedHeart/Windows/Debug.cs:
```diff
@@ -181,18 +195,76 @@ private static void DrawDebugStatusInfo()
+            ImGui.SameLine();
+            ImGui.TextColored(ImGuiColors.ParsedGreen, coordText);
+
+            ImGui.SetCursorPosX(ImGui.GetCursorPosX());
```
何意味

VeeverSW replied:
debug还是复制粘贴遗留了没注意.jpg

AtmoOmen commented on UntarnishedHeart/Windows/Debug.cs:
```diff
@@ -181,18 +195,76 @@ private static void DrawDebugStatusInfo()
+        {
+            ImGui.TextColored(ImGuiColors.DalamudRed, "无法获取摄像机信息");
+            return;
+        }
```
直接 return

VeeverSW committed:
1. KeyState内部改为枚举值
2. 将DateTime改为Environment.TickCount64来减少开销
3. DrawCursorToWorld删除超出游戏边界提示
4. 将寻路控制按钮改为公用，放到了运行状态下面

VeeverSW committed:
fix:
1. 更改寻路控制位置到执行模式下方
2. 删除PathFindHelper.cs，调用OmenTools的PathFindHelper
3. Notification 全都用Chat替代

AtmoOmen commented on UntarnishedHeart/Configuration.cs:
无实质修改, 删去

AtmoOmen commented on UntarnishedHeart/Windows/Main.cs:
```diff
@@ -79,6 +80,11 @@ public override void Draw()
+
+                ImGui.Separator();
+                ImGui.Spacing();
+
```
放在模式配置后

AtmoOmen commented on UntarnishedHeart/Utils/PathFindHelper.cs:
删除，改为使用 OmenTools 中的

AtmoOmen reviewed (changes requested):
Notification 输出的警告错误全部改成聊天输出

VeeverSW committed:
Merge branch 'main' of https://github.com/VeeverSW/UntarnishedHeart

# Conflicts:
#	UntarnishedHeart/Windows/Main.cs   resolved by main version
